lib/moment-range.js

```diff
@@ -327,6 +320,339 @@ DateRange.prototype.diff = function(unit) {
+  var actualStart = this.actualStart;
+  var actualEnd = this.actualEnd;
+    
+  if(upperLimit < actualStart || actualEnd < lowerLimit) {
+    //if does not overlap
+    this.start = this.end = undefined;    
+  } 
+  else {
+    //if overlaps
+    if (actualStart <= lowerLimit) {
+      this.start = lowerLimit;
+    } else {
+      this.start = actualStart;
+    }
+
+    if (actualEnd <= upperLimit) {
+      this.end = actualEnd;
+    } else {
+      this.end = upperLimit; 
+    }
+  }
+  this._setDuration();
+};
+
+ /**
+ * sets start to new value
+ * null or undefined values will clear the date
+ * 
+ * @param {date string|date object|moment object} - start
+ * 
+ * @return this.start 
+ * @return this.end
+ * 
+ */
+DateRange.prototype.setStart = function(start) {
+ this.set(start, this.actualEnd);
+}
+
+ /**
+ * sets end to new value
+ * null or undefined values will clear the date
+ * 
+ * @param {date string|date object|moment object} - end
+ * 
+ * @return this.start 
+ * @return this.end
+ * 
+ */
+DateRange.prototype.setEnd = function(end) {
+  this.set(this.actualStart, end);
+}
+
+ /**
+ * sets lowerlimit to new value
+ * null or undefined values will clear the date
+ * 
+ * @param {date string|date object|moment object} - lowerlimit
+ * 
+ * @return this.start 
+ * @return this.end
+ * 
+ */
+DateRange.prototype.setLowerLimit = function(lowerLimit) {
+  this.setLimit(lowerLimit, this.upperLimit);
+}
+
+ /**
+ * sets upperlimit to new value
+ * 
+ * @param {date string|date object|moment object} - upperlimit
+ * 
+ * @return this.start 
+ * @return this.end
+ * 
+ */
+DateRange.prototype.setUpperLimit = function(upperLimit) {
+  this.setLimit(this.lowerLimit, upperLimit);
+}
+
+ /**
+ * clears start and/or end values
+ * Undefined parameters clear both.
+ * 
+ * @param {boolean} - clearStart
+ * @param {boolean} - clearEnd
+ * 
+ * @return this.start 
+ * @return this.end
+ * 
+ */
+DateRange.prototype.clear = function(clearStart, clearEnd) {
+  var start = clearStart === false ? this.actualStart : undefined;
+  var end = clearEnd === false ? this.actualEnd : undefined;
+  this.set(start, end);
+}
+
+ /**
+ * clears upper and/or lower limts
+ * 
+ * @param {boolean} - clear lower limit
+ * @param {boolean} - clear upper limit
+ * 
+ * @return this.start 
+ * @return this.end
+ * 
+ */
+DateRange.prototype.clearLimits = function(clearLowerLimit, clearUpperLimit) {
+  var lowerLimit = clearLowerLimit === false ? this.lowerLimit : undefined;
+  var upperLimit = clearUpperLimit === false ? this.upperLimit : undefined;
+  this.setLimit(lowerLimit, upperLimit);
+}
+
+/** 
+ * Shifts start and end values forward
+ *  If no limits are set the time will shift indefinitely
+ * 
+ * @param {number}
+ * 
+ * @return this.start
+ * @return this.end
+ */
+DateRange.prototype.shiftForward = function(duration) {
+  var start = this.actualStart + duration;
+  var end = this.actualEnd + duration;
+  this.set(start, end);
```

Comment:
this is not considering the case for when duration is undefined.
create a unit test